### PR TITLE
wgsl: Implement `determinant` validation tests

### PR DESCRIPTION
**Pull request by zoddicus**
This only tests the shapes of the inputs compilation, and not that later validation passes, since this builtin has undefined error.

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
```
Comment (dneto0) [inline src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts:11]: nit: This was a bit hard to read. I'd suggest adding a comment

> // Generate a dictionary mapping each matrix type variation (columns,rows, floating point type) to a nontrivial matrix value of that type.

File: src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      t.selectDeviceOrSkipTestCase('shader-f16');
+    }
+  })
+  .fn(t => {
+    const cols = t.params.cols;
+    const rows = t.params.rows;
+    const type = t.params.type;
+    const arg = kMatrixCases[`mat${cols}x${rows}_${type}`];
+    t.expectCompileResult(
+      cols === rows,
```
Comment (dneto0) [inline src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts:57]: Nice.

**Review by dneto0 — APPROVED** [review]
Thanks!

File: src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/validation/expression/call/builtin/determinant.spec.ts:11]: Done

**Pull request merged**